tests/transports/smpp/test_client.py

```diff
@@ -100,6 +115,8 @@ async def test_start(client: EsmeClient, server_stream, nursery):
 
 async def test_send_response_pdu(client: EsmeClient, server_stream):
     """Response PDUs should not wait for a reply"""
+    await test_start(client, server_stream)
```

Comment:
We don't actually want to test the startup every time, merely start the client. Maybe extract that (without the assertions) into a helper?

At some point it'll probably make sense to have a fake SMSC implementation that we can use instead of talking to the server stream directly.

Reply:
Yeah I also thought of using SMPPSim for this, but that would mean that you need java installed to run the tests.

I was deciding between just having the `tcp_server`, with helpers to receive and assert on inbound, and build outbound, traffic. Or building a fake SMSC implementation that automatically responds the way an SMSC would.